### ISSUE TITLE
return PUC first in search results if PUC is searched on

### ISSUE THREAD

**Issue by kdionisio** [HumanExposure/factotum_elastic#24]
As a Factotum user
I want to see the PUC link first in search results
If I search the name of a PUC

e.g., if I search on 'personal care', the following search results are returned in the 'PUC' tab of the elastic search results, in this order:
Personal care -- sexual wellness --
Personal care -- eye care and contacts -- contact care
Landscape/Yard -- plants and garden -- garden care
Personal care -- hair styling and care -- ethnic hair care
Landscape/Yard -- plants and garden -- tree care products
Vehicle -- boat care and maintenance -- boat cleaner - hull care
Vehicle -- boat care and maintenance -- boat cleaner - deck care
Personal care -- body care set --
Cleaning products and household care --

see search results here: http://factotum.epa.gov/search/puc/?q=cGVyc29uYWwgY2FyZQ==

Rather, the overarching 'Personal care' PUC should be returned first, with all other 'Personal care - XX - XX' PUCs returned in search results before something like a 'Vehicle' PUC

Definition of Done:
In elastic search results, specifically in the PUC tab, if the name of a PUC is searched, that PUC(s) should be returned first in search results

**Comment by fang-yan-csra** [comment]
Elastic has its algorithm to calculate the score and return search results in relevancy. It seems like "garden care" got higher rank because it appears in a short puc_prodtype (only 2 words).

There is a way to change the boost of a field to make puc_gencat more important. However, it's hard to decide "how much is the proper weigh on each field?" Based on my testing, set puc_gencat boost to 10 makes this field dominating.

Another way is to use query_string (instead of free text search) to accept lucene query syntax, like "personal AND care" to narrow the search down to only return documents containing the phrase "personal care". Not sure if this will solve the use case here.

**Comment by mwfrost** [comment]
Currently, `gen_cat`, `prod_fam`, and `prod_type` are all text fields in the search index, not keyword fields. That means a search for "personal care" will return matches when _either_ word appears in any of the three fields. So the first PUC result is 198: "Personal care -- hair styling and care -- ethnic hair care". The phrase "personal care" is the `gen_cat`, and the word "care" appears in the `prod_fam`, the `prod_type`, and the `description`. There are abundant reasons that result should be at the very top.

As long as we keep tokenizing the three PUC fields, PUC 137, which is the top of the "Personal Care" tree, will always score lower than one of its children in a search for "personal care" if that child has "care" in its `prod_fam` or `prod_type` fields. One way to fix this would be to use exclusively keyword types for these three fields, so that "care" would not match any PUCs, and "personal care" would match 137 and all its children. There is probably a way to sort by score then by PUC ID. 

Switching to keywords means that only searching for "make-up and related" exactly as we store it would return PUC 213 and its children: "makeup and related" or "makeup" would return the four PUCs that use "makeup" in the `prod_type` field. It would also return any PUCs where "makeup" appears in a description, since `description` is still a tokenized text field, not a keyword field.

Short version: it is tough to promote a grandparent PUC above its grandchildren in the search results when the grandchildren have two additional fields that could potentially match the search term, thereby seeming more relevant.

**Comment by mwfrost** [comment]
Proposed solution: instead of using elasticsearch to return a list of PUCs that has been reassembled from the search index, we simply render the PUC records, filtered by the search term and ordered by their normal sort sequence.

**Comment by mwfrost** [comment]
Another alternative would be to use a `datatable` in `puc_list.html`, and add a filter field like we use on the other pages.
